File: Minerador/Paulo/events/event_type.py
from enum import Enum, auto

class EventType(Enum):
    # Gerenciamento de Grafo
    GRAPH_CREATE = auto()
    GRAPH_ADD_VERTEX = auto()
    GRAPH_ADD_EDGE = auto()
    GRAPH_REMOVE_EDGE = auto()
    GRAPH_HAS_EDGE = auto()
    GRAPH_GET_VERTEX_COUNT = auto()
    GRAPH_GET_EDGE_COUNT = auto()
    GRAPH_SET_VERTEX_WEIGHT = auto()
    GRAPH_GET_VERTEX_WEIGHT = auto()
    GRAPH_SET_EDGE_WEIGHT = auto()
    GRAPH_GET_EDGE_WEIGHT = auto()
    GRAPH_IN_DEGREE = auto()
    GRAPH_OUT_DEGREE = auto()
    GRAPH_IS_SUCCESSOR = auto()
    GRAPH_IS_PREDECESSOR = auto()
    GRAPH_IS_DIVERGENT = auto()
    GRAPH_IS_CONVERGENT = auto()
    GRAPH_IS_INCIDENT = auto()
    GRAPH_IS_CONNECTED = auto()
    GRAPH_IS_EMPTY = auto()
    GRAPH_IS_COMPLETE = auto()
    GRAPH_EXPORT_GEPHI = auto()

    # Algoritmos
    ALGO_BFS = auto()
    ALGO_DFS = auto()
    ALGO_SHORTEST_PATH = auto()
    ALGO_TOPOLOGICAL_SORT = auto()
    ALGO_STRONGLY_CONNECTED = auto()

    # Métricas de Centralidade
    METRIC_DEGREE_CENTRALITY = auto()
    METRIC_BETWEENNESS_CENTRALITY = auto()
    METRIC_CLOSENESS_CENTRALITY = auto()
    METRIC_PAGERANK = auto()

    # Métricas de Estrutura
    METRIC_DENSITY = auto()
    METRIC_CLUSTERING_COEFFICIENT = auto()
    METRIC_ASSORTATIVITY = auto()

    # Métricas de Comunidade
    METRIC_COMMUNITY_DETECTION = auto()
    METRIC_BRIDGING_TIES = auto()

    # Mineração de Dados (GitHub)
    MINING_LOAD_CSV = auto()
    MINING_BUILD_GRAPH1_COMMENTS = auto()
    MINING_BUILD_GRAPH2_CLOSURES = auto()
    MINING_BUILD_GRAPH3_REVIEWS = auto()
    MINING_BUILD_INTEGRATED_GRAPH = auto()

    # Respostas
    RESPONSE_SUCCESS = auto()
    RESPONSE_ERROR = auto()

File: Paulo_Code/Prototipo_java_single-thread/src/main/java/br/pucminas/grafo/events/EventType.java
package br.pucminas.grafo.events;

/**
 * Catálogo central de todos os tipos de eventos da arquitetura EDA.
 * Cada tipo representa uma "chamada de API" interna.
 */
public enum EventType {

    // ── Gerenciamento de Grafo ──────────────────────────────────────────────
    GRAPH_CREATE,
    GRAPH_ADD_VERTEX,
    GRAPH_ADD_EDGE,
    GRAPH_REMOVE_EDGE,
    GRAPH_HAS_EDGE,
    GRAPH_GET_VERTEX_COUNT,
    GRAPH_GET_EDGE_COUNT,
    GRAPH_SET_VERTEX_WEIGHT,
    GRAPH_GET_VERTEX_WEIGHT,
    GRAPH_SET_EDGE_WEIGHT,
    GRAPH_GET_EDGE_WEIGHT,
    GRAPH_IN_DEGREE,
    GRAPH_OUT_DEGREE,
    GRAPH_IS_SUCCESSOR,
    GRAPH_IS_PREDECESSOR,
    GRAPH_IS_DIVERGENT,
    GRAPH_IS_CONVERGENT,
    GRAPH_IS_INCIDENT,
    GRAPH_IS_CONNECTED,
    GRAPH_IS_EMPTY,
    GRAPH_IS_COMPLETE,
    GRAPH_EXPORT_GEPHI,

    // ── Algoritmos ─────────────────────────────────────────────────────────
    ALGO_BFS,
    ALGO_DFS,
    ALGO_SHORTEST_PATH,
    ALGO_TOPOLOGICAL_SORT,
    ALGO_STRONGLY_CONNECTED,

    // ── Métricas de Centralidade ────────────────────────────────────────────
    METRIC_DEGREE_CENTRALITY,
    METRIC_BETWEENNESS_CENTRALITY,
    METRIC_CLOSENESS_CENTRALITY,
    METRIC_PAGERANK,

    // ── Métricas de Estrutura ───────────────────────────────────────────────
    METRIC_DENSITY,
    METRIC_CLUSTERING_COEFFICIENT,
    METRIC_ASSORTATIVITY,

    // ── Métricas de Comunidade ──────────────────────────────────────────────
    METRIC_COMMUNITY_DETECTION,
    METRIC_BRIDGING_TIES,

    // ── Mineração de Dados (GitHub) ─────────────────────────────────────────
    MINING_LOAD_CSV,
    MINING_BUILD_GRAPH1_COMMENTS,
    MINING_BUILD_GRAPH2_CLOSURES,
    MINING_BUILD_GRAPH3_REVIEWS,
    MINING_BUILD_INTEGRATED_GRAPH,

    // ── Respostas ───────────────────────────────────────────────────────────
    RESPONSE_SUCCESS,
    RESPONSE_ERROR
}
